src/components/tests/SourcesTree.spec.js

```diff
@@ -85,4 +86,622 @@ describe("SourcesTree", () => {
+        .find(".sources-list")
+        .simulate("keydown", { keyCode: 13 });
+      expect(selectLocation).toHaveBeenCalledWith({
+        sourceId: "server1.conn13.child1/39"
+      });
+    });
+
+    it("does not select if no item is focused on", async () => {
+      await component
+        .find(".sources-list")
+        .simulate("keydown", { keyCode: 13 });
+      expect(selectLocation).not.toHaveBeenCalled();
+    });
+  });
+
+  describe("handles items", () => {
+    const { component, props } = render({
+      setExpandedState: jest.fn()
+    });
+
+    it("getChildren from directory", async () => {
+      const item = createMockItem("http://mdn.com/views", "views", ["a", "b"]);
+      const children = component
+        .find("ManagedTree")
+        .props()
+        .getChildren(item);
+      expect(children).toEqual(["a", "b"]);
+    });
+
+    it("getChildren from non directory", async () => {
+      const children = component
+        .find("ManagedTree")
+        .props()
+        .getChildren(itemOne);
+      expect(children).toEqual([]);
+    });
+
+    it("onExpand", async () => {
+      const expandedState = ["x", "y"];
+      await component
+        .find("ManagedTree")
+        .props()
+        .onExpand({}, expandedState);
+      expect(props.setExpandedState).toHaveBeenCalledWith(expandedState);
+    });
+
+    it("onCollapse", async () => {
+      const expandedState = ["y", "z"];
+      await component
+        .find("ManagedTree")
+        .props()
+        .onCollapse({}, expandedState);
+      expect(props.setExpandedState).toHaveBeenCalledWith(expandedState);
+    });
+
+    it("getParent", async () => {
+      const item = component.state("sourceTree").contents[0].contents[0];
+      const parent = component
+        .find("ManagedTree")
+        .props()
+        .getParent(item);
+
+      expect(parent.path).toEqual("mdn.com");
+      expect(parent.contents).toHaveLength(3);
+    });
+  });
+
+  describe("getPath", () => {
+    const { component } = render({
+      sources: I.Map({
+        "server1.conn13.child1/39": createMockSource(
+          "server1.conn13.child1/39",
+          "http://mdn.com/one.js"
+        ),
+        "server1.conn13.child1/59": createMockSource(
+          "server1.conn13.child1/59",
+          "http://mdn.com/blackboxed.js",
+          true
+        )
+      })
+    });
+
+    it("should return path for item", async () => {
+      const path = component.instance().getPath(itemOne);
+      expect(path).toEqual("http://mdn.com/one.js/one.js/");
+    });
+
+    it("should return path for blackboxedboxed item", async () => {
+      const item = createMockItem(
+        "http://mdn.com/blackboxed.js",
+        "blackboxed.js",
+        I.Map({ id: "server1.conn13.child1/59" })
+      );
+      const path = component.instance().getPath(item);
+      expect(path).toEqual("http://mdn.com/blackboxed.js/blackboxed.js/update");
+    });
+  });
+
+  describe("test redux connect", () => {
```

Comment:
you shouldnt need to test redux (we trust redux to work and have its own tests, it is a dependancy). We should be able to assert that this works correctly from how the component works in the other tests.

This level of "in-component-testing" is very brittle, and makes the tests hard to maintain.

Reply:
this was more to get coverage on `mapToProps`. I don't really need to test each function, just that mapToProps is being called. Or export `mapToProps` and mock it and test it's been called.